desktop-app/app/components/Addressinput/index.js

```diff
@@ -175,6 +224,64 @@ class AddressBar extends React.Component<Props> {
     }
     return address;
   };
+
+  _addUrlToExistingSearchResult = () => {
+
+    let existingSearchResults = getExistingSearchResults();
+    let formattedUrl = this._normalize(this.state.userTypedAddress);
+
+    if(existingSearchResults?.length){
+     let updatedSearchResults = [...existingSearchResults];
+
+     const index = updatedSearchResults.findIndex(eachSearchResult => eachSearchResult.url === formattedUrl);
+
+     index!== (undefined|| -1 || null) ? updatedSearchResults[index].visitedCount = updatedSearchResults[index].visitedCount+1 :
+            updatedSearchResults.push({url: formattedUrl,visitedCount:1})
+
+     addUrlToSearchResults(updatedSearchResults);
+
+    }
+
+    else {
+      let addNewUrl = [];
+        addNewUrl.push({
+          url: formattedUrl,
+          visitedCount: 1
+        });
+      addUrlToSearchResults(addNewUrl);
+    }
+    this.setState({
+      finalUrlResult:[]
+    })
+  }
+
+  _sortedExistingUrlSearchResult = (filteredData) => { //Most visited site should appear first in the list
+    filteredData.sort((a, b)=> {
+       if(a.visitedCount > b.visitedCount){
+         return -1
+       }
+       else if(a.visitedCount < b.visitedCount){
+         return 1
+       }
+       return 0;
+    });
+
+     return filteredData;
+
+  }
+
+  _filterExistingUrl = () => {
+    const filteredData = filter(getExistingSearchResults(), (eachResult) => eachResult.url.toLowerCase().includes(this.state.userTypedAddress));
```

Comment:
Let's not read the data from settings forever time, let's read it once and maintain the list in memory. 

Just write a copy to settings whenever a new URL has to be added.